File: src/main/java/su/foxochat/dto/gateway/response/TypingStartDTO.java

package su.foxochat.dto.gateway.response;

import lombok.Getter;
import lombok.Setter;
import su.foxochat.constant.GatewayConstant;

import java.util.Map;

@Getter
@Setter
public class TypingStartDTO {

	private int op;

	private Map<String, Long> d;

	public TypingStartDTO(long channelId, long userId, long timestamp) {
		this.op = GatewayConstant.Opcode.HELLO.ordinal();
		this.d = Map.of("channel_id", channelId, "user_id", userId, "timestamp", timestamp);
	}
}

File: src/main/java/su/foxogram/dto/gateway/response/TypingStartDTO.kt
package su.foxogram.dto.gateway.response

import lombok.Getter
import lombok.Setter
import su.foxogram.constant.GatewayConstant
import java.util.Map

@Getter
@Setter
class TypingStartDTO(channelId: Long, userId: Long, timestamp: Long) {
    private val op: Int = GatewayConstant.Opcode.HELLO.ordinal

    private val d: MutableMap<String, Long>? =
        Map.of<String, Long>("channel_id", channelId, "user_id", userId, "timestamp", timestamp)

}
